code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -145,7 +145,10 @@
 	if(!status)
 		to_chat(user, "<span class='notice'>Secure the igniter first!</span>")
 		return
-	to_chat(user, "<span class='notice'>You ignite [src]!</span>")
+	if(!lit)
+		to_chat(user, "<span class='notice'>You ignite [src]!</span>")
```

Comment:
remove the if and else and do
`to_chat(user, "<span class='notice'>You [lit ? "extinguish" : "ignite"] [src]!</span>")`